source/AMDSClientUi.cpp

```diff
@@ -0,0 +1,332 @@
+
+	statusLabel_ = new QLabel(tr("This examples requires that you run the Acquaman Data Server example as well."));
+
+	connect(hostLineEdit_, SIGNAL(textChanged(QString)), this, SLOT(enableRequestDataButton()));
+	connect(portLineEdit_, SIGNAL(textChanged(QString)), this, SLOT(enableRequestDataButton()));
+
+	/// ==== buttons section ====
+	connectServerButton_ = new QPushButton(tr("Connect to Server"));
+	connectServerButton_->setDefault(true);
+
+	requestDataButton_ = new QPushButton(tr("Request Data"));
+	quitButton_ = new QPushButton(tr("Quit"));
+
+	buttonBox_ = new QDialogButtonBox;
+	buttonBox_->addButton(connectServerButton_, QDialogButtonBox::ActionRole);
+	buttonBox_->addButton(requestDataButton_, QDialogButtonBox::ActionRole);
+	buttonBox_->addButton(quitButton_, QDialogButtonBox::RejectRole);
+
+	connect(connectServerButton_, SIGNAL(clicked()), this, SLOT(connectToServer()));
+	connect(requestDataButton_, SIGNAL(clicked()), this, SLOT(sendClientRequest()));
+	connect(quitButton_, SIGNAL(clicked()), this, SLOT(close()));
+
+	/// ==== message compose section ====
+	requestTypeComboBox_ = new QComboBox;
+	requestTypeComboBox_->addItem("Introspect");
+	requestTypeComboBox_->addItem("Statistics");
+	requestTypeComboBox_->addItem("Start Time + Count");
+	requestTypeComboBox_->addItem("Relative Count + Count");
+	requestTypeComboBox_->addItem("Start Time to End Time");
+	requestTypeComboBox_->addItem("Middle Time + Count Before to Count After");
+	requestTypeComboBox_->addItem("Continuous");
+	requestTypeComboBox_->setCurrentIndex(0);
+
+	const QStandardItemModel* model = qobject_cast<const QStandardItemModel*>(requestTypeComboBox_->model());
+	for(int x = 1; x < 7; x++){
+		QStandardItem* item = model->item(x);
+		item->setFlags(item->flags() & ~(Qt::ItemIsSelectable|Qt::ItemIsEnabled));
+		// visually disable by greying out - works only if combobox has been painted already and palette returns the wanted color
+		item->setData(requestTypeComboBox_->palette().color(QPalette::Disabled, QPalette::Text), Qt::TextColorRole);
+	}
+
+	bufferNameListView_ = new QListView();
+	QStringList bufferNames("All");
+	resetBufferListView(bufferNames);
+
+	time1Edit_ = new QDateTimeEdit(QDateTime::currentDateTime());
+	time1Edit_->setDisplayFormat("HH:mm:ss.zzz");
+	time2Edit_ = new QDateTimeEdit(QDateTime::currentDateTime());
+	time2Edit_->setDisplayFormat("HH:mm:ss.zzz");
+	count1Edit_ = new QLineEdit();
+	count2Edit_ = new QLineEdit();
+	includeStatusDataCheckbox_ = new QCheckBox();
+	enableFlattenDataCheckbox_ = new QCheckBox();
+
+	activeContinuousConnectionComboBox_ = new QComboBox;
+	activeContinuousConnectionComboBox_->setFixedWidth(200);
+	activeContinuousConnectionComboBox_->addItem("");
+
+	resultsEdit_ = new QTextEdit();
+
+	connect(requestTypeComboBox_, SIGNAL(currentIndexChanged(QString)), this, SLOT(onRequestTypeChanged(QString)));
+
+
+//	amptekIndex = new QSpinBox();
```

Comment:
This is from the original codes. I'm not sure about the usage or it yet. Will leave it there for a little while